php/flights.php

```diff
@@ -257,17 +252,4 @@
     );
 }
 
-if ($export == "gcmap") {
-    // Output the redirect URL.
-    header("Location: http://www.gcmap.com/mapui?P=" . implode(',', $rows) . "&MS=bm");
-    return;
-}
-
-if (count($rows)) {
-    $separator = ($export == "export" || $export == "backup")
-        ? "\r\n"
-        // Not for $export == "gcmap"
-        : "\n";
-
-    echo implode($separator, $rows);
-}
+echo implode("\r\n", $rows);
```

Comment:
Not that there really should be much difference for most (decent) applications of `\r\n` vs `\n`, but...

https://github.com/jpatokal/openflights/commit/0f1adfcaae5d35ddb7337761da480d4fd34bec77#diff-21a477194502a985c437c81d564ce284eaff75d08a36e2d7e6ad6a72f017bcd6L175-L179

Do we pass anything else as a parameter value to `export=` other than export or backup?

Reply:
I don't think we pass other values to `export` but we do handle requests that don't set `export` (for "list flights"). I didn't see any breakage but the `\r` (we split on newlines in JS) would be invisible in the browser.

I'll revert this part of the change. This part will have to change anyway for us to send JSON back to the UI.

Reply:
https://github.com/jpatokal/openflights/blame/899b1c53a53b1094448d12bf202092898e8024a9/php/flights.php#L181-L186

https://github.com/jpatokal/openflights/commit/cc8dfd79bfd6a83795b43bc0f9c1c69d8b7f7d55#diff-8c9cf639f05f3287fbb6ef307938f6a66297a4081b8eafbadc996dee50315fafR865

```
var rows = str.split("\n");
```

Yeah, I still see 3 `str.split("\n")`; `updateFilter`, `updateMap` and `listFlights` in openflights.js now.

Leaving trailing `\r` isn't the best, and like you say, when #1376 happens, this gets fixed properly anyway, so we don't need to worry about it too much!

Reply:
Fixed in the rebase/force push.